src/qt/walletmodel.cpp

```diff
@@ -853,6 +942,33 @@ bool WalletModel::isSpent(const COutPoint& outpoint) const
     return wallet->IsSpent(outpoint.hash, outpoint.n);
 }
 
+void WalletModel::listCoins(std::map<ListCoinsKey, std::vector<ListCoinsValue>>& mapCoins, bool fTransparent) const
+{
+    if (fTransparent) {
+        listCoins(mapCoins);
+    } else {
+        listAvailableNotes(mapCoins);
+    }
+}
+
+void WalletModel::listAvailableNotes(std::map<ListCoinsKey, std::vector<ListCoinsValue>>& mapCoins) const
+{
+    std::vector<SaplingNoteEntry> notes;
+    Optional<libzcash::SaplingPaymentAddress> dummy = nullopt;
+    wallet->GetSaplingScriptPubKeyMan()->GetFilteredNotes(notes, dummy);
+    for (const auto& note : notes) {
+        ListCoinsKey key{QString::fromStdString(KeyIO::EncodePaymentAddress(note.address)), false, nullopt};
+        ListCoinsValue value{
+            note.op.GetHash(),
```

Comment:
`BaseOutPoint::GetHash()` returns the hash of the serialized outpoint (including both `BaseOutPoint::hash` and `BaseOutPoint::n`).
Instead, we only want to pass directly `BaseOutPoint::hash` here. 
```suggestion
            note.op.hash,
```

Reply:
ah fuck, love you 💜 . I owe you a good whisky.

Reply:
pushed under your authorship. Really nice catch, sneaky one.